python/cuml/internals/array.py

```diff
@@ -1172,12 +1172,16 @@ def from_input(
         if (
             not fail_on_order and order != arr.order and order != "K"
         ) or make_copy:
-            arr = cls(
-                arr.mem_type.xpy.array(
-                    arr.to_output("array"), order=order, copy=make_copy
-                ),
-                index=index,
-            )
+            if make_copy:
+                data = arr.mem_type.xpy.array(
+                    arr.to_output("array"), order=order
+                )
+            else:
+                data = arr.mem_type.xpy.asarray(
+                    arr.to_output("array"), order=order
+                )
```

Comment:
It seems odd to check this above and here. Is there a better way to write this so we only check `make_copy` once? Perhaps with an `elif`?

Reply:
Probably, I was working on adding polars support (which needs to happen here), so I'm reviewing the whole conditionals... would you mind opening an issue to cleanup this so it doesn't block/slow down this PR?

Reply:
For clarity tried to include some suggestions below. Though have no strong feelings on whether they are included